packages/babel-preset-babili/src/index.js

```diff
@@ -1,133 +1,98 @@
+  ["memberExpressions",   require("babel-plugin-transform-member-expression-literals"),    true],
+  ["mergeVars",           require("babel-plugin-transform-merge-sibling-variables"),       true],
+  ["numericLiterals",     require("babel-plugin-minify-numeric-literals"),                 true],
+  ["propertyLiterals",    require("babel-plugin-transform-property-literals"),             true],
+  ["regexpConstructors",  require("babel-plugin-transform-regexp-constructors"),           true],
+  ["removeConsole",       require("babel-plugin-transform-remove-console"),                false],
+  ["removeDebugger",      require("babel-plugin-transform-remove-debugger"),               false],
+  ["removeUndefined",     require("babel-plugin-transform-remove-undefined"),              true],
+  ["replace",             require("babel-plugin-minify-replace"),                          true],
+  ["simplify",            require("babel-plugin-minify-simplify"),                         true],
+  ["simplifyComparisons", require("babel-plugin-transform-simplify-comparison-operators"), true],
+  ["typeConstructors",    require("babel-plugin-minify-type-constructors"),                true],
+  ["undefinedToVoid",     require("babel-plugin-transform-undefined-to-void"),             true]
 ];
 
+const PROXIES = {
+  keepFnName: ["mangle", "deadcode"],
+  keepClassName: ["mangle", "deadcode"]
+};
+
 module.exports = preset;
 
 function preset(context, _opts = {}) {
   const opts = isPlainObject(_opts) ? _opts : {};
 
-  // to track every plugin is used
-  const usedPlugins = new Set();
-
-  const optionsMap = PLUGINS.map(plugin =>
-    option(plugin[0], plugin[1], plugin[2])
-  ).reduce((acc, cur) => {
-    Object.defineProperty(acc, cur.name, {
-      get() {
-        usedPlugins.add(cur.name);
-        return cur;
-      }
-    });
-    return acc;
-  }, {});
-
-  const optionsTree = group(
-    "options",
-    [
-      optionsMap.evaluate,
-      optionsMap.deadcode,
-
-      group("unsafe", [
-        optionsMap.flipComparisons,
-        optionsMap.simplifyComparisons,
-        optionsMap.guards,
-        optionsMap.typeConstructors
-      ]),
-
-      optionsMap.infinity,
-      optionsMap.mangle,
-      optionsMap.numericLiterals,
-      optionsMap.replace,
-      optionsMap.simplify,
-      optionsMap.builtIns,
-
-      group("properties", [
-        optionsMap.consecutiveAdds,
-        optionsMap.memberExpressions,
-        optionsMap.propertyLiterals
-      ]),
-
-      optionsMap.mergeVars,
-      optionsMap.booleans,
-      optionsMap.undefinedToVoid,
-      optionsMap.regexpConstructors,
-
-      optionsMap.removeConsole,
-      optionsMap.removeDebugger,
-      optionsMap.removeUndefined,
-
-      proxy("keepFnName", [optionsMap.mangle, optionsMap.deadcode]),
+  // validate options
+  const validOptions = [...PLUGINS.map(p => p[0]), ...Object.keys(PROXIES)];
+  for (let name in opts) {
```

Comment:
The accessor is `opts[name]` and `opts` is user passed. So better to validate all that we access instead of just accessing the own properties.